MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -230,6 +230,16 @@ open class RouteLegProgress: NSObject {
     public func isCurrentStep(_ step: RouteStep) -> Bool {
         return leg.steps.index(of: step) == stepIndex
     }
+    
+    /*
+     Return bool whether step provided is the upcoming `Step`.
+     */
+    public func isUpComingStep(_ step: RouteStep) -> Bool {
+        guard 1 + stepIndex < leg.steps.count else {
+            return false
+        }
+        return leg.steps.index(of: step) == 1 + stepIndex
```

Comment:
`index(of:)` performs a linear search through the array, but I think this method and `isCurrentStep(_:)` are both unnecessary. We could get the same result by checking `step == leg.steps[stepIndex + 1]`. In fact, the call site could just say `step == routeController.routeProgress.currentLegProgress.currentStep`.